File: examples/run-actor.js
// Node.js example using the official apify-client package (CommonJS).
// Install:  npm install apify-client
// Usage:    APIFY_TOKEN=your_token node run-actor.js

const { ApifyClient } = require('apify-client');

const client = new ApifyClient({
    token: process.env.APIFY_TOKEN,
});

async function main() {
    const input = {
        discoveryMode: 'seedList',
        seedList: ['acme.com', 'Beta Consulting LLC', 'https://gamma-industries.example'],
        maxLeads: 25,
        includeIntentScore: true,
        skipKnownLeads: true,
    };

    console.log('Starting actor-18-b2b-lead-magnet run...');
    const run = await client.actor('5QufcYxRkFNHM4h8K').call(input);

    console.log(`Run finished with status: ${run.status}`);

    const { items } = await client.dataset(run.defaultDatasetId).listItems();

    console.log(`Fetched ${items.length} leads:`);
    for (const item of items) {
        console.log(`- ${item.recipient_or_defendant_name} (${item.website}) intent=${item.intentScore ?? 'n/a'}`);
    }
}

main().catch((err) => {
    console.error('Actor run failed:', err);
    process.exit(1);
});

File: examples/run_actor.py
"""
Python example using the official apify-client package.
Install: pip install apify-client
Usage:   APIFY_TOKEN=your_token python run_actor.py
"""

import os

from apify_client import ApifyClient

client = ApifyClient(os.environ["APIFY_TOKEN"])

run_input = {
    "discoveryMode": "seedList",
    "seedList": ["acme.com", "Beta Consulting LLC", "https://gamma-industries.example"],
    "maxLeads": 25,
    "includeIntentScore": True,
    "skipKnownLeads": True,
}

print("Starting actor-18-b2b-lead-magnet run...")
run = client.actor("5QufcYxRkFNHM4h8K").call(run_input=run_input)

print(f"Run finished with status: {run['status']}")

dataset_items = client.dataset(run["defaultDatasetId"]).list_items().items

print(f"Fetched {len(dataset_items)} leads:")
for item in dataset_items:
    intent = item.get("intentScore", "n/a")
    print(f"- {item.get('recipient_or_defendant_name')} ({item.get('website')}) intent={intent}")
